docs/index.adoc

```diff
@@ -359,6 +359,100 @@ entirely, they can save a transaction and gas by calling
 `TBTCVault.unmintAndRedeem` which routes into authorizing the Bridge to
 <<redeeming,redeem>> via `bank.approveBalanceAndCall`.
 
+=== Maintainers
+
+`Bridge` has a set of public knowledge functions that need to be called from
+time to time. A public-knowledge function is where anyone has access to the
+information required to assemble the transaction and the transaction does not
+lead to punishment (slashing).
+
+To compensate for calling these transactions, the caller has the gas spent
+reimbursed by a DAO-funded ETH pool in the same transaction. This works only if
+callers follow the order and do not try to front-run each other. Given that the
+reimbursement code is constructed in such a way that the caller is net-zero or
+minimally positive after the refund, there should be no reason for callers to
+front-run each other. On the other hand, a single malicious front-runner can
+break this scheme. If there is someone who does not follow the order and keeps
+submitting transactions - for which they are reimbursed - everyone else will
+stop doing that because they will only burn gas on reverted transactions.
+
+One approach to address this problem could be to enforce the submission order
+on-chain. Although it sounds easy at first, it is complicated and expensive in
+practice. The code needs to take into account scenarios when someone is
+temporarily offline and missed their submission round or when someone is
+permanently offline and keeps missing their round all the time which causes
+delays and slows down the network. Attempts to address these problems by marking
+someone as ineligible for rewards in case they missed their round are making the
+on-chain submission order enforcement mechanism only more expensive.
+
+Another approach is to not enforce the order on-chain, have everyone with at
+least the minimum stake be able to submit public-knowledge transactions, and
+expect that everyone respects the informal agreement about the submission order.
+This approach works fine as long as there is no single malicious player with the
+minimum stake to break this scheme. To address this problem, we could introduce
+a mechanism for slashing submitters not respecting the order but it is again,
+complex in practice, and makes us reimplement solutions like Gelato Network.
+
+We approach this problem in the easiest way possible by reducing a problem to
+a smaller group and introducing a list of submitters who are getting reimbursed
+for their calls. We call them maintainers. Maintainers are third-party bots who
+are expected to follow an off-chain informal agreement about the submission
+order and do not front-run each other. If they do, they can get deauthorized for
+reimbursements by the DAO. Maintainers are no more trusted than any other
+address on Ethereum. Anyone can submit public-knowledge `Bridge` transactions
```

Comment:
We could consider adding somewhere here an information about the fact that maintainers are getting authorized by DAO. It's mentioned later ("Only
DAO-authorized maintainers can call `Bridge`...), but it would make sense to say they're authorized before we mention deauthorizing. 
Also, what do you think about mentioning the names of the functions authorizing/deauthorizing the maintainers?

Reply:
Good idea! See 561ce4ccba8a153feabb24c5cfe6b63b04c9f4df please.